prog/encodingexec_test.go

```diff
@@ -151,6 +151,15 @@ func TestSerializeForExec(t *testing.T) {
 				instrEOF,
 			},
 		},
+		{
+			"syz_test$bufconst(&(0x7f0000000000)={\"776c616e3000\", 0x43})",
+			[]uint64{
+				instrCopyin, dataOffset + 0, argData, 6, 0x00306e616c77,
+				instrCopyin, dataOffset + 8, argConst, 2, 0x43,
```

Comment:
Looking at this code, I found a latent bug in sys/decl.go. I've just pushed a fix:
https://github.com/google/syzkaller/commit/4ae183a7833b09bb8699161335b8402e8de69322
Previously we assumed that StrConstType has size and alignment 8. Previously it has no effect as StrConstType always used with indirection.
You make it possible to make it a field in structs. And the wrong size/align cause bug in struct alignment padding calculations.

Please pull/rebase and then change dataOffset + 8 to dataOffset + 6.

Actually please also remove 1 byte from the string, so that its size is 5, then we will also test alignment padding. I.e. string is 5 bytes, but the next int16 field should start at offset 6 due to alignment requirements. As the result the serialized program should look as:

```
				instrCopyin, dataOffset + 0, argData, 5, 0x00306e616c,
				instrCopyin, dataOffset + 6, argConst, 2, 0x43,
``